### PR TITLE
Chore: added Server Instances endpoint types

### PR DESCRIPTION
**Pull request by felipe-rod123**
## Proposed changes (including videos or screenshots)

Created typing for endpoint definitions on `instances.ts`.

### CODE REVIEW

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
looks like the typings for an "Instance" object is really messed up 🙈 

if you look the [endpoint code](https://github.com/RocketChat/Rocket.Chat/blob/d42f1643f1b6ceff55ae079509879a4433741ffb/app/api/server/v1/instances.ts#L20), it returns 
```
{
  ...instance,
  connection,
}
```

`instance` is `IInstanceStatus` - but `IInstanceStatus` is missing a lot of properties on the type definition
`connection` is the return of [getConnection](https://github.com/RocketChat/Rocket.Chat/blob/d42f1643f1b6ceff55ae079509879a4433741ffb/server/stream/streamBroadcast.js#L296) which is essentially an `IInstance` object.

maybe this PR is a good opportunity to get all this types fixed? 👀

File: app/api/server/v1/instances.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ API.v1.addRoute(
 
 			return API.v1.success({
 				instances: instances.map((instance: IInstanceStatus) => {
-					const connection = getInstanceConnection(instance);
+					const connection = getInstanceConnection(instance) as IInstanceStatus;
```
Comment (ggazzo) [inline app/api/server/v1/instances.ts:20]: create a `d.ts` definition

File: definition/IInstance.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (ggazzo) [inline definition/IInstance.ts:1]: delete the file

File: definition/IInstanceStatus.ts [hunk 3]
```diff
@@ -3,4 +3,28 @@ export interface IInstanceStatus {
 	extraInformation?: {
 		port?: number;
 	};
+
+	// IInstance props
```
Comment (ggazzo) [inline definition/IInstanceStatus.ts:7]:
```suggestion
```

**Review by ggazzo — CHANGES_REQUESTED** [review]
Please check the comments ;)

**Review by ggazzo — APPROVED** [review]

**Pull request merged**